### ISSUE TITLE
Same rendering for barrier=ditch and walls

### ISSUE THREAD

**Issue by jeisenbe** [gravitystorm/openstreetmap-carto#3975]
### Expected behavior
- The tag `barrier=ditch` is used to tag dry linear depressions which are normally not filled with water, except perhaps during flooding, and instead serve as a barrier to travel 
- Such ditches are barriers to most motor vehicles, but are usually only a minor inconvenience to pedestrians, equestrians and bicyclists, similar to an intermittent `waterway=ditch` or `waterway=stream`. They may also be passible for off-road vehicles.
- Therefore, it is reasonable to expect that `barrier=ditch` will be rendered differently than `barrier=wall` and `barrier=fence`, which are significant vertical barriers, normally not passable to pedestrians, equestrians or cyclists.
- There should be some subtle similarity between `waterway=ditch` + `intermittent=yes` and `barrier=ditch` - because ditch barriers will often have running water at least during heavy rains or floods. 

[Tag:barrier=ditch](https://wiki.openstreetmap.org/wiki/Tag:barrier=ditch)

### Actual behavior
- The tag barrier=ditch renders identically to vertical barriers including `barrier=wall` and `barrier=fence` and `barrier=retaining_wall`
- There is no similarity between the rendering of `waterway=ditch` + `intermittent=yes` and `barrier=ditch`

### Link illustrating the problem

Ditches in Libya (without rendering of parallel `barrier=embankments`...)
https://www.openstreetmap.org/#map=16/31.9764/10.6907
![barrier-embankment-after](https://user-images.githubusercontent.com/42757252/68585255-16742400-04c5-11ea-8b8e-de6d7d6bf415.png)

**Comment by imagico** [comment]
Note in a related problem the combination of waterways with barrier=ditch is still rendered in a confusing fashion:

https://www.openstreetmap.org/way/447594080

IIRC this was meant to be fixed but was not due to an operator precedence error in SQL code.

I would be against rendering barrier=ditch in a way that implies the presence of water.

**Comment by jeisenbe** [comment]
> I would be against rendering barrier=ditch in a way that implies the presence of water.

Agreed. I was thinking of trying a dark thin line, similar to other barriers, but using the same dash pattern as intermittent ditches - though that dash pattern should probably be improved (longer dashes, shorter gaps). It might also help to add a "halo" on each side, like with ditches.

> The combination of waterways with barrier=ditch is still rendered in a confusing fashion:

That's surprising! In the `line-barriers` layer we have this limitation:

```
WHERE barrier IN ('chain', 'city_wall', 'embankment', 'ditch', 'fence', 'guard_rail',
                  'handrail', 'hedge', 'kerb', 'retaining_wall', 'wall')
              OR historic = 'citywalls'
              AND (waterway IS NULL OR waterway NOT IN ('river', 'canal', 'stream', 'drain', 'ditch'))
```

Can we fix it by moving the OR statement to the end like this:
```
              AND (waterway IS NULL OR waterway NOT IN ('river', 'canal', 'stream', 'drain', 'ditch'))
              OR historic = 'citywalls'
```
Or is it better to use parentheses?

**Comment by jeisenbe** [comment]
> the combination of waterways with barrier=ditch is still rendered in a confusing fashion:

Can we fix it by moving the OR statement to the end like this:
```
              AND (waterway IS NULL OR waterway NOT IN ('river', 'canal', 'stream', 'drain', 'ditch'))
              OR historic = 'citywalls'
```
Or is it better to use parentheses?

**Comment by imagico** [comment]
> Or is it better to use parentheses?

I think it is.

**Comment by GunSmoker** [comment]
If this would be fixed - then barrier=furrow is probably should be rendered as barrier=ditch (or similar to it)??

barrier=furrow is used to indicate (for example) firebreak furrows - it is not exactly a ditch, as it is just removed top layer of soil. Typically it is also a path.

![Polosa-mineralizirovannaya](https://user-images.githubusercontent.com/3242603/84499811-fa836b80-acbb-11ea-8da3-39373d595a53.jpg)

**Comment by Adamant36** [comment]
> If this would be fixed - then barrier=furrow is probably should be rendered as barrier=ditch (or similar to it)??

The numbers are a little low for it to be rendered at this point. In the meantime it would be cool if there was at least a wiki article for it though. Although, my guess is that it's a redundant tag due to them being extremely similar to ditches. For instance both are used for irrigation. While I know farrows are not as deep as ditches, I'd argue the distinction is superficial and that it's impossible to tell "deepness" from satellite images or even on the ground to a degree. Plus, where would the deepness cutoff for it becoming a ditch be anyway? If there is one, there probably isn't a way to figure it out. At least not by the average (or most) mappers.
